presto-manager-tests/src/main/java/com/teradata/prestomanager/tests/TestBasicFunction.java

```diff
@@ -0,0 +1,314 @@
+    {
+        Response response = client
+                .target(UriBuilder
+                        .fromUri(controllerUri)
+                        .path("/presto/stop")
+                        .queryParam("stopType", "TERMINATE")
+                        .queryParam("scope", "ALL")
+                        .build())
+                .request(TEXT_PLAIN)
+                .buildPost(Entity.text(""))
+                .invoke();
+
+        assertEquals(response.getStatus(), 207);
+    }
+
+    private void uninstallPresto(Client client)
+    {
+        Response response = client
+                .target(UriBuilder
+                        .fromUri(controllerUri)
+                        .path("/package")
+                        .queryParam("scope", "ALL")
+                        .build())
+                .request(TEXT_PLAIN)
+                .buildDelete()
+                .invoke();
+
+        assertEquals(response.getStatus(), 207);
+    }
+
+    private boolean isInstalled()
+    {
+        List<JsonObject> statusList = parseStatusResponse(queryPrestoStatus());
+        if (statusList.size() != numPrestoNodes) {
+            return false;
+        }
+        for (JsonObject status : statusList) {
+            if (!status.get("installed").getAsBoolean()) {
+                return false;
+            }
+        }
+        return true;
+    }
+
+    private boolean isRunning()
+    {
+        List<JsonObject> statusList = parseStatusResponse(queryPrestoStatus());
+        if (statusList.size() != numPrestoNodes) {
+            return false;
+        }
+        for (JsonObject status : statusList) {
+            if (!status.get("running").getAsBoolean()) {
+                return false;
+            }
+        }
+        return true;
+    }
+
+    private List<JsonObject> parseStatusResponse(Response response)
+    {
+        JsonParser jsonParser = new JsonParser();
+        JsonObject prestoStatus = jsonParser.parse(response.readEntity(String.class)).getAsJsonObject();
+
+        // Get the list of nodeIDs
+        List<String> nodeIdList = new ArrayList<>();
+        for (Map.Entry<String, JsonElement> entry : prestoStatus.entrySet()) {
+            nodeIdList.add(entry.getKey());
+        }
+
+        List<JsonObject> statusList = new ArrayList<>();
+        for (String nodeId : nodeIdList) {
+            statusList.add(prestoStatus.getAsJsonObject(nodeId).getAsJsonObject("body"));
+        }
+        return statusList;
+    }
+
+    private Response queryPrestoStatus()
+    {
+        return client
+                .target(UriBuilder
+                        .fromUri(controllerUri)
+                        .path("/presto/status")
+                        .queryParam("scope", "ALL")
+                        .build())
+                .request(APPLICATION_JSON)
+                .buildGet()
+                .invoke();
+    }
+
+    private void retry(TestRunnable runnable, boolean expectedResult, int retryTime, int timeout)
```

Comment:
also call this assertWithRetry() and throw an assertion error if you fail.  (maybe pass in a failure message too?)